lib/plugins.js

```diff
@@ -33,12 +33,20 @@ module.exports.load = function ({ config }) {
       }
 
       // register schema and validate settings
+      let validate;
       try {
-        schemas.register('plugin', pluginName, plugin.schema)(settings);
+        validate = schemas.register('plugin', pluginName, plugin.schema);
       } catch (e) {
         logger.error(`Wrong schema definition for ${pluginName}`, e);
       }
 
+      if (validate !== undefined) {
+        const {isValid, error} = validate(settings);
+        if (!isValid) {
+          throw new Error('Failed to validate settings: ' + error);
```

Comment:
Nip: please use ES6 literal strings.
Also, make sure to log the error (`logger.error`)